test/types/virtuals.test.ts

```diff
@@ -118,5 +118,18 @@ function autoTypedVirtuals() {
   const doc = new TestModel();
   expectType<string>(doc.domain);
 
-  expectType<FlatRecord<AutoTypedSchemaType & VirtualsType >>({} as InferredDocType);
+  TestModel.findOne({}).then((doc) => {
+    if (doc) {
+      expectType<string>(doc.domain);
+      const json = doc.toJSON();
+      expectType<string>(json.domain);
```

Comment:
This is incorrect. `domain` shouldn't be in `json` unless you have `toJSON: { virtuals: true }` in your schema or globally.